### PR TITLE
add extra rex parser

### PR DESCRIPTION
**Pull request by jamesgorrie**
We seem to own all of them.

### CODE REVIEW

File: common-lib/src/main/scala/com/gu/mediaservice/lib/cleanup/SupplierProcessors.scala [hunk 1]
```diff
@@ -245,11 +245,14 @@ object ReutersParser extends ImageProcessor {
...
+  val SlashRex = ".+/ Rex Features".r
+
+
+  def apply(image: Image): Image = (image.metadata.source, image.metadata.credit) match {
     // TODO: cleanup byline/credit
-    case Some("Rex Features") => image.copy(
-      usageRights = Agency("Rex Features")
-    )
+    case (Some("Rex Features"), _) => image.copy(usageRights = rexAgency)
+    case (_, Some(SlashRex()))     => image.copy(usageRights = rexAgency)
```
Comment (theefer) [inline common-lib/src/main/scala/com/gu/mediaservice/lib/cleanup/SupplierProcessors.scala]: Note you can use `case foo | bar =>` if you want.

File: common-lib/src/test/scala/com/gu/mediaservice/lib/cleanup/SupplierProcessorsTest.scala [hunk 2]
```diff
@@ -385,6 +385,13 @@ class SupplierProcessorsTest extends FunSpec with Matchers with MetadataHelper {
       processedImage.metadata.credit should be(Some("Tim Ireland/REX Shutterstock"))
       processedImage.metadata.source should be(Some("Rex Features"))
     }
+
+    it("should match '*/ Rex Features' credit") {
+      val image = createImageFromMetadata("credit" -> "Bleddyn Butcher / Rex Features")
+      val processedImage = applyProcessors(image)
+
+      processedImage.usageRights should be (Agency("Rex Features"))
```
Comment (theefer) [inline common-lib/src/test/scala/com/gu/mediaservice/lib/cleanup/SupplierProcessorsTest.scala]: Do we want to save the prefix as the collection, or is it other random things (byline etc)?

File: common-lib/src/main/scala/com/gu/mediaservice/lib/cleanup/SupplierProcessors.scala [hunk 3]
```diff
@@ -245,11 +245,14 @@ object ReutersParser extends ImageProcessor {
 }
 
 object RexParser extends ImageProcessor {
-  def apply(image: Image): Image = image.metadata.source match {
+  val rexAgency = Agency("Rex Features")
+  val SlashRex = ".+/ Rex Features".r
```
Comment (theefer) [inline common-lib/src/main/scala/com/gu/mediaservice/lib/cleanup/SupplierProcessors.scala]: Is there always a space before `Rex` (e.g. in Getty there isn't, but Rex may be different)?

**Comment by theefer** [comment]
:+1:

File: common-lib/src/main/scala/com/gu/mediaservice/lib/cleanup/SupplierProcessors.scala (hunk 1 above)
Comment (jamesgorrie) [inline common-lib/src/main/scala/com/gu/mediaservice/lib/cleanup/SupplierProcessors.scala]: We're matching on source and credit here, hence the tuple. Unless I'm missing something?

File: common-lib/src/test/scala/com/gu/mediaservice/lib/cleanup/SupplierProcessorsTest.scala (hunk 2 above)
Comment (jamesgorrie) [inline common-lib/src/test/scala/com/gu/mediaservice/lib/cleanup/SupplierProcessorsTest.scala]: I know we're meant to keep the same credit.
The prefix can be a photographer, collection or agency, so I thought it best nt to assume we can figure that out (I remember trying that a while ago and almost going :rage4:).

File: common-lib/src/main/scala/com/gu/mediaservice/lib/cleanup/SupplierProcessors.scala (hunk 3 above)
Comment (jamesgorrie) [inline common-lib/src/main/scala/com/gu/mediaservice/lib/cleanup/SupplierProcessors.scala]: Yip, again, trying not to be presumptive, from what I've seen and been shown, this is always the case.

File: common-lib/src/main/scala/com/gu/mediaservice/lib/cleanup/SupplierProcessors.scala (hunk 1 above)
Comment (theefer) [inline common-lib/src/main/scala/com/gu/mediaservice/lib/cleanup/SupplierProcessors.scala]: I meant I think you can write a single `case` as `case (Some("Rex Features"), _) | (_, Some(SlashRex()))` if you prefer. I don't mind particularly.

File: common-lib/src/test/scala/com/gu/mediaservice/lib/cleanup/SupplierProcessorsTest.scala (hunk 2 above)
Comment (theefer) [inline common-lib/src/test/scala/com/gu/mediaservice/lib/cleanup/SupplierProcessorsTest.scala]: :+1:

File: common-lib/src/main/scala/com/gu/mediaservice/lib/cleanup/SupplierProcessors.scala (hunk 1 above)
Comment (jamesgorrie) [inline common-lib/src/main/scala/com/gu/mediaservice/lib/cleanup/SupplierProcessors.scala]: Ah, never knew that, I'll give it a go.

**Pull request merged**